sqlmesh/core/context.py

```diff
@@ -1319,51 +1372,60 @@ def _model_tables(self) -> t.Dict[str, str]:
         If a snapshot has not been versioned yet, its view name will be returned.
         """
         return {
-            name: snapshot.table_name()
+            snapshot.fqn: snapshot.table_name()
             if snapshot.version
             else snapshot.qualified_view_name.for_environment(
                 EnvironmentNamingInfo(
                     name=c.PROD, suffix_target=self.config.environment_suffix_target
                 )
             )
-            for name, snapshot in self.snapshots.items()
+            for snapshot in self.snapshots
         }
 
     def _snapshots(
         self, models_override: t.Optional[UniqueKeyDict[str, Model]] = None
-    ) -> t.Dict[str, Snapshot]:
+    ) -> t.Dict[SnapshotId, Snapshot]:
         prod = self.state_reader.get_environment(c.PROD)
         remote_snapshots = (
             {
-                snapshot.name: snapshot
+                snapshot.snapshot_id: snapshot
                 for snapshot in self.state_reader.get_snapshots(prod.snapshots).values()
             }
             if prod
             else {}
         )
+        remote_snapshot_model_fqn_to_snapshot = {
+            snapshot.fqn: snapshot for s_id, snapshot in remote_snapshots.items()
+        }
+        models = models_override or self._models
+        local_nodes: t.Dict[str, t.Union[Model, StandaloneAudit]] = {
+            **models,
+            **self._standalone_audits,
+        }
 
-        local_nodes = {**(models_override or self._models), **self._standalone_audits}
         nodes = local_nodes.copy()
         audits = self._audits.copy()
         projects = {config.project for config in self.configs.values()}
 
-        for name, snapshot in remote_snapshots.items():
+        for snapshot in remote_snapshots.values():
+            name = snapshot.fqn
             if name not in nodes and snapshot.node.project not in projects:
                 nodes[name] = snapshot.node
-
                 if snapshot.is_model:
                     for audit in snapshot.audits:
                         if name not in audits:
                             audits[name] = audit
 
-        def _nodes_to_snapshots(nodes: t.Dict[str, Node]) -> t.Dict[str, Snapshot]:
-            snapshots: t.Dict[str, Snapshot] = {}
+        def _nodes_to_snapshots(nodes: t.Dict[str, Node]) -> t.Dict[SnapshotId, Snapshot]:
+            snapshots: t.Dict[SnapshotId, Snapshot] = {}
             fingerprint_cache: t.Dict[str, SnapshotFingerprint] = {}
 
             for node in nodes.values():
-                if node.name not in local_nodes and node.name in remote_snapshots:
-                    snapshot = remote_snapshots[node.name]
-                    ttl = snapshot.ttl
+                if (
+                    node.fqn not in local_nodes
+                    and node.fqn in remote_snapshot_model_fqn_to_snapshot
+                ):
+                    ttl = remote_snapshot_model_fqn_to_snapshot[node.fqn].ttl
```

Comment:
why did you remove the snapshot = line?

Reply:
It doesn't do anything right? The following line sets the snapshot.

Reply:
doesn't it pull from remote_snapshots?

Reply:
So this: 
```
snapshot = remote_snapshots[node.name]
ttl = snapshot.ttl
```
was converted to this:
```
ttl = remote_snapshot_model_fqn_to_snapshot[node.fqn].ttl
```

Both pull from remote snapshot right? I just skip the step of assigned the remote snapshot to `snapshot` since it doesn't do anything due to the fact we assign `snapshot` to a new value on the following line.